wezterm-gui/src/overlay/copy.rs

```diff
@@ -1141,14 +1160,73 @@ impl Pane for CopyOverlay {
                 (KeyCode::Char(c), KeyModifiers::NONE)
                 | (KeyCode::Char(c), KeyModifiers::SHIFT) => {
                     // Type to add to the pattern
-                    render.pattern.push(c);
+                    render.search_line.insert_char(c);
+
+                    render.schedule_update_search();
+                }
+                (KeyCode::Char('H'), KeyModifiers::CTRL)
+                | (KeyCode::Backspace, KeyModifiers::NONE) => {
+                    render
+                        .search_line
+                        .kill_text(Movement::BackwardChar(1), Movement::BackwardChar(1));
+
                     render.schedule_update_search();
                 }
-                (KeyCode::Backspace, KeyModifiers::NONE) => {
-                    // Backspace to edit the pattern
-                    render.pattern.pop();
+                (KeyCode::Delete, KeyModifiers::NONE) => {
+                    render
+                        .search_line
+                        .kill_text(Movement::ForwardChar(1), Movement::None);
+
                     render.schedule_update_search();
                 }
+                (KeyCode::Backspace, KeyModifiers::ALT)
+                | (KeyCode::Char('W'), KeyModifiers::CTRL) => {
+                    render
+                        .search_line
+                        .kill_text(Movement::BackwardWord(1), Movement::BackwardWord(1));
+
+                    render.schedule_update_search();
+                }
+                (KeyCode::Backspace, KeyModifiers::SUPER) => {
+                    render
+                        .search_line
+                        .kill_text(Movement::StartOfLine, Movement::StartOfLine);
+
+                    render.schedule_update_search();
+                }
+                (KeyCode::Char('K'), KeyModifiers::CTRL) => {
+                    render
+                        .search_line
+                        .kill_text(Movement::EndOfLine, Movement::EndOfLine);
+
+                    render.schedule_update_search();
+                }
+                (KeyCode::Char('B'), KeyModifiers::CTRL)
+                | (KeyCode::ApplicationLeftArrow, KeyModifiers::NONE)
+                | (KeyCode::LeftArrow, KeyModifiers::NONE) => {
+                    render.search_line.exec_movement(Movement::BackwardChar(1));
+                }
+                (KeyCode::Char('F'), KeyModifiers::CTRL)
+                | (KeyCode::ApplicationRightArrow, KeyModifiers::NONE)
+                | (KeyCode::RightArrow, KeyModifiers::NONE) => {
+                    render.search_line.exec_movement(Movement::ForwardChar(1));
+                }
+                (KeyCode::Char('b'), KeyModifiers::CTRL)
```

Comment:
FWIW, `ctrl-b` is potentially ambiguous with `ctrl-B` on unix systems due to https://github.com/wez/wezterm/issues/1906

In my shell ctrl-b (either case for `b`) is equivalent to `BackwardChar(1)`

Of the actions you've added here that use CTRL-ALPHA where you have upper and lower case versions, which are actually winning and resolving for you?

I'd be inclined to remove the ambiguous case from these mappings without good evidence that things are working well and consistently across platforms.

Reply:
I've removed potentially ambiguous (at least from my point of view). What do you think now? Is any of the mappings can cause conflicts?

Reply:
ah, what I meant was, rather than having both the uppercase and lowercase versions of things like ctrl-B, with different actions, just pick one of those actions as there is potential for ambiguity.
I did kinda like that you had `ctrl-a` `ctrl-e` for start/end of line and would like to see that come back!

Reply:
Ok) Let's sync then.
CTRL-H - On Mac OS it conflicts with OS key binding. Leave or remove?
CTRL-K - again, conflict with Mac OS. Leave or remove?
CTRL-W - probably we can keep it.
CTRL-B, CTRL-F - initially I mapped it BackwardChar/ForwardChar. You wanted to remove ambiguous, so I can delete those mappings.
CTRL-b, CTRL-f - BackwardWord/ForwardWord. Keep it. Do you want to add CTRL-B, CTRL-F for BackwardWord/ForwardWord too?
CTRL-a, CTRL-e - StartOfLine/EndOfLine. Again, do you want me to map CTRL-A, CTRL-E too? Or keep only capital or only lowercase?

I think, it is worth making it configurable, but I would like to do it in another pr.

Reply:
CTRL-H - keep
CTRL-K - keep
CTRL-W - keep
CTRL-B, CTRL-F - BackwardChar/ForwardChar. keep.
CTRL-b, CTRL-f - remove as it could conflict with the uppercase version, and the char rather than word behavior is not what I see in eg: zsh's line editor for these key combinations.
CTRL-a, CTRL-e - StartOfLine/EndOfLine. keep, but make A/E uppercase.

Reply:
Done.